dbms/src/Flash/Coprocessor/DAGQueryBlockInterpreter.cpp

```diff
@@ -762,10 +762,11 @@ void DAGQueryBlockInterpreter::executeWindow(
     DAGPipeline & pipeline,
     WindowDescription & window_description)
 {
+    executeExpression(pipeline, window_description.before_window);
+
     /// If there are several streams, we merge them into one
     executeUnion(pipeline, max_streams, log);
```

Comment:
For the window function, are the input streams always 1?
Is executeUnion required here?
Or `RUNTIME_ASSERT(pipeline.size() == 1, log, "xxxxxx")`

Reply:
if window query have no partition or order,  the source of window will be receiver. In this case, the stream may be not 1.
```sql
explain select *,row_number() over() from window_function_0;
+-------------------------------+---------+--------------+-------------------------+-----------------------------------------------------------------------+
| id                            | estRows | task         | access object           | operator info                                                         |
+-------------------------------+---------+--------------+-------------------------+-----------------------------------------------------------------------+
| TableReader_13                | 9955.00 | root         |                         | data:ExchangeSender_12                                                |
| └─ExchangeSender_12           | 9955.00 | mpp[tiflash] |                         | ExchangeType: PassThrough                                             |
|   └─Window_7                  | 9955.00 | mpp[tiflash] |                         | row_number()->Column#7 over(rows between current row and current row) |
|     └─ExchangeReceiver_11     | 9955.00 | mpp[tiflash] |                         |                                                                       |
|       └─ExchangeSender_10     | 9955.00 | mpp[tiflash] |                         | ExchangeType: PassThrough                                             |
|         └─TableFullScan_9     | 9955.00 | mpp[tiflash] | table:window_function_0 | keep order:false                                                      |
+-------------------------------+---------+--------------+-------------------------+-----------------------------------------------------------------------+
```

Reply:
ok